packages/scratch-vm/src/extensions/scratch3_musiccreation/musiccreationhelpers.js

```diff
@@ -342,94 +345,171 @@ class MusicCreationHelpers {
-        const musicState = this._getMusicState(util.target);
-        const inst = musicState.currentInstrument;
         const instrumentInfo = this.INSTRUMENT_INFO[inst];
         const sampleArray = instrumentInfo.samples;
         const sampleIndex = this._selectSampleIndexForNote(note, sampleArray);
-
         // If the audio sample has not loaded yet, bail out
-        if (typeof this._instrumentPlayerArrays[inst] === 'undefined') return;
-        if (typeof this._instrumentPlayerArrays[inst][sampleIndex] === 'undefined') return;
-
-        // Fetch the sound player to play the note.
-        const engine = util.runtime.audioEngine;
+        if (typeof this._instrumentPlayerArrays[inst] === 'undefined' ||
+            typeof this._instrumentPlayerArrays[inst][sampleIndex] === 'undefined') {
+            console.log('uninitialized instruments');
+            return null;
+        }
 
         if (!this._instrumentPlayerNoteArrays[inst][note]) {
             this._instrumentPlayerNoteArrays[inst][note] = this._instrumentPlayerArrays[inst][sampleIndex].take();
         }
 
-        const player = this._instrumentPlayerNoteArrays[inst][note];
+        const player = this._getPlayer(inst,note);
 
-        if (player.isPlaying && !player.isStarting) {
-            // Take the internal player state and create a new player with it.
-            // `.play` does this internally but then instructs the sound to
-            // stop.
-            player.take();
+        return {player:player, 
+                data:
+                    {instInfo: instrumentInfo, 
+                     sampleArray: sampleArray, 
+                     sampleIndex: sampleIndex,
+                     note: note,
+                     duration: dur}};
+    }
+
+    /**
+     * Plays the note given by @param noteInfo and recursively
+     * sets up an event chain to play the rest of the notes in @param seq
+     * @param {object} noteInfo - element of @param seq containing
+     *                       'note', 'index', and 'beats' fields.
+     * @param {Array[]} seq - array of objects containing information about a note and its duration
+     * @param {BlockUtility} util 
+     * @param {number} l - length of @param seq 
+     * @private
+     * @augments @param util's stackFrame.duration to be 0 once the last note in @param seq 
+     *           has stopped playing. 
+     */
+    _playNoteFromSeq (noteInfo, seq, util,l, inst) {
+        const i = noteInfo['index'];
+        const last = i === l-1;
+        if (this._concurrencyCounter > this.CONCURRENCY_LIMIT) return;
+        const playerAndData = this.createPlayer(util,noteInfo['note'],noteInfo['duration'], inst);
+        if (!playerAndData) {
+            console.log(`null data for note ${noteInfo}`);
+            return;
+        }
+        const player = playerAndData['player'];
+        player.once('stop', () => {
+            console.log(`stopped note ${i+1}`);
```

Comment:
Can we drop these logs?

Error logs are fine to keep in for now, but best to keep the console as clear as possible